tuf/repository_lib.py

```diff
@@ -1980,14 +2019,15 @@ def write_metadata_file(metadata, filename, version_number,
  
   if consistent_snapshot:
     dirname, basename = os.path.split(written_filename)
-    
     basename = basename.split(METADATA_EXTENSION, 1)[0]
     version_and_filename = str(version_number) + '.' + basename + METADATA_EXTENSION 
     written_consistent_filename = os.path.join(dirname, version_and_filename)
 
-    logger.info('Linking ' + repr(written_consistent_filename))
-    os.link(written_filename, written_consistent_filename)
+    #logger.info('Linking ' + repr(written_consistent_filename))
+    #os.link(written_filename, written_consistent_filename)
```

Comment:
When making changes, always hard linking old versions to the current will result in all previous versions (1.root, 2.root, etc) pointing to the current (root.json). 

Writing the consistent version and then linking the current to that (e.g. linking root.json -> 3.root.json) should work, but requires more changes because now "write current" doesn't happen in every case. Holding two copies is a small constant increase in storage (#versions -> #versions + 1)

Either solution is fine, I left the commented code to remind myself about the issue and clearly forgot to come back to it!